content-src/components/DiscoveryStreamComponents/DSLinkMenu/DSLinkMenu.jsx

```diff
@@ -0,0 +1,72 @@
+import {FormattedMessage, injectIntl} from "react-intl";
+import {LinkMenu} from "content-src/components/LinkMenu/LinkMenu";
+import React from "react";
+
+export class _DSLinkMenu extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeCard: null,
+      showContextMenu: false,
+    };
+    this.onMenuButtonClick = this.onMenuButtonClick.bind(this);
+    this.onMenuUpdate = this.onMenuUpdate.bind(this);
+    this.onMenuShow = this.onMenuShow.bind(this);
+    this.contextMenuButtonRef = React.createRef();
+  }
+
+  onMenuButtonClick(event) {
+    event.preventDefault();
+    this.setState({
+      activeCard: this.props.index,
+      showContextMenu: true,
+    });
+  }
+
+  onMenuUpdate(showContextMenu) {
+    if (!showContextMenu) {
+      const dsLinkMenuHostDiv = this.contextMenuButtonRef.current.parentElement;
+      dsLinkMenuHostDiv.parentElement.classList.remove("active", "last-item");
+    }
+    this.setState({showContextMenu});
+  }
+
+  onMenuShow() {
+    const dsLinkMenuHostDiv = this.contextMenuButtonRef.current.parentElement;
+    if (window.scrollMaxX > 0) {
```

Comment:
I'm still seeing scrollMax, is that still needed with the opacity check?

Reply:
I think I imagined rendering an opacity 0 menu, grabbing the width, and checking that the width + starting position is less than the window viewport.

Reply:
Opacity check is helping with hiding the possible jank due to flip from default right direction to left as context menu renders with opacity 0 and force scroll and thats why scrollMaxX check.